src/backend/metal/src/device.rs

```diff
@@ -835,7 +848,21 @@ impl hal::Device<Backend> for Device {
                                 *old = Some(((new.0).0.clone(), new.1));
                             }
                         },
-                        (&Sampler(_), _) | (&SampledImage(_), _) => panic!("mismatched descriptor set type"),
+                        (&UniformBuffer(ref buffers), Some(&mut n::DescriptorSetBinding::ConstantBuffer(ref mut vec))) |
+                        (&StorageBuffer(ref buffers), Some(&mut n::DescriptorSetBinding::ConstantBuffer(ref mut vec))) => {
```

Comment:
This should be a separate DescriptorSetBinding (`StorageBuffer`), which needs to be defined on set allocation afaik.